### DISCUSSION TITLE
[i18n] Locale routing is broken on /{locale}/[param] if param is a locale.

### DISCUSSION THREAD

**Discussion by noidar** [vercel/next.js#72926] [Help]
### Summary

Hello,

I am experiencing an issue with locale-based routing in Next.js when the param in the URL is a locale itself. Specifically, the routing fails when the URL follows the pattern /{locale}/[param], where the param is the same as the locale.

Example URLs that fail to route correctly:

http://localhost:3000/en/en/details
http://localhost:3000/de/de/details

I am attempting to implement a solution using this repository as a reference: [i18-country-language-issue](https://github.com/noidar/i18-country-language-issue).

Additionally, I found related discussions on this issue, but no solutions have been provided so far:

[Next.js Issue #31420](https://github.com/vercel/next.js/issues/31420)
[Next.js Issue #21714](https://github.com/vercel/next.js/issues/21714)
Has anyone encountered a similar issue or found a workaround? Any insights or solutions would be greatly appreciated.

Thank you!

**Comment by felipepalazzo** [comment]
Hey @noidar, I'm experiencing the exact same issue. I'm using Next 13.4.13 and next-translate 2.5.3. Have you managed to find a fix for that?

**Reply by noidar** [reply]
I haven't found a solution yet :( , so this is on hold. Previously, I was able to redirect in middleware.

**Reply by noidar** [reply]
@felipepalazzo  here is a working example with page rout https://github.com/noidar/i18n-routing-page-issue

**Reply by felipepalazzo** [reply]
hi @noidar, thanks for sharing. I managed to overcome this issue by doing a rewrite. Something like: 

```js
      {
        source: '/:locale/:countryCode([A-Za-z]{2})',
        destination: '/:locale/splash',
        locale: false,
      },
```
